src/map.js

```diff
@@ -92,50 +92,52 @@ function PluginContainer() {
         }
     }
 
-    function loadMap(config) {
+    async function loadMap(config) {
+        const systemSettings = await fetchSystemSettings([
+            'keyBingMapsApiKey',
+            'keyDefaultBaseMap',
+        ]);
         if (config.id && !isUnmounted(config.el)) {
-            // Load favorite
-            mapRequest(config.id)
-                .then(renameBoundaryLayerToOrgUnitLayer)
-                .then(favorite =>
-                    loadLayers({
-                        ...config,
-                        ...favorite,
-                    })
-                );
+            mapRequest(config.id, systemSettings.keyDefaultBaseMap).then(
+                favorite =>
+                    loadLayers(
+                        {
+                            ...config,
+                            ...favorite,
+                        },
+                        systemSettings.keyBingMapsApiKey
+                    )
+            );
+        } else if (!config.mapViews) {
+            getConfigFromNonMapConfig(
+                config,
+                systemSettings.keyDefaultBaseMap
+            ).then(config =>
+                loadLayers(config, systemSettings.keyBingMapsApiKey)
+            );
         } else {
-            translateConfig(config)
-                .then(renameBoundaryLayerToOrgUnitLayer)
-                .then(loadLayers);
+            loadLayers(
+                getMigratedMapConfig(config, systemSettings.keyDefaultBaseMap),
+                systemSettings.keyBingMapsApiKey
+            );
         }
     }
 
-    async function loadLayers(config) {
+    async function loadLayers(config, keyBingMapsApiKey) {
         if (!isUnmounted(config.el)) {
-            let basemap = config.basemap || 'osmLight';
+            const basemap = config.basemap;
 
-            // Default basemap is required, visibility is set to false below
-            if (basemap === 'none') {
-                basemap = 'osmLight';
-            }
-
-            const basemapId = basemap.id || basemap;
-
-            if (isValidUid(basemapId)) {
-                const externalLayer = await getExternalLayer(basemapId);
-                basemap = {
-                    id: basemapId,
-                    config: {
-                        type: TILE_LAYER,
-                        ...externalLayer,
-                    },
-                };
+            if (isValidUid(config.basemap.id)) {
+                const externalLayer = await getExternalLayer(config.basemap.id);
+                basemap.config = createExternalLayerConfig(externalLayer);
             } else {
-                basemap = defaultBasemaps().find(map => map.id === basemapId);
+                basemap.config = defaultBasemaps().find(
+                    map => map.id === config.basemap.id
+                ).config;
```

Comment:
I'm getting an error when testing http://localhost:8082/plugin.html?id=zDP78aJU8nX

_Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'config')
    at loadLayers (map.js:135)_

There are stored maps that don't have any basemap property, example: 
https://play.dhis2.org/dev/api/37/maps/zDP78aJU8nX.json

The code will break here as config.basemap.id = undefined

When no basemap is set for a saved map, the fallback should be the default basemap - this seems to work fine in the app.

Reply:
I'm getting the same error is basemap is "none"

Reply:
Strange. That url of the map the doesn't have a basemap.- works fine for me.